### PR TITLE
refactor(rm): Move to clap + add a test

### PR DESCRIPTION
**Pull request by sylvestre**
Will probably fail on some configurations because https://github.com/uutils/coreutils/pull/1629/commits/cc0a94ef7b486ce8a5c1554a82dcdfbb537b2ddd
needs to be merged first

### CODE REVIEW

**Comment by rivy** [comment]
> Will probably fail on some configurations because [cc0a94e](https://github.com/uutils/coreutils/commit/cc0a94ef7b486ce8a5c1554a82dcdfbb537b2ddd)
> needs to be merged first

I got the same error; the fix is already included in the 'uucore' merge. Once that's merged, this can be rebased on top and it should work.

**Comment by sylvestre** [comment]
I landed it here:
https://github.com/uutils/coreutils/pull/1633
to avoid the pain in the CI

File: src/uu/rm/src/rm.rs [hunk 1]
```diff
@@ -40,83 +41,139 @@ struct Options {
...
+static OPT_FORCE: &str = "force";
+static OPT_PROMPT: &str = "prompt";
+static OPT_PROMPT_MORE: &str = "prompt-more";
+static OPT_INTERACTIVE: &str = "interactive";
+static OPT_ONE_FILE_SYSTEM: &str = "one-file-system";
+static OPT_NO_PRESERVE_ROOT: &str = "no-preserve-root";
+static OPT_PRESERVE_ROOT: &str = "preserve-root";
+static OPT_RECURSIVE: &str = "recursive";
+static OPT_DIR: &str = "dir";
+static OPT_VERBOSE: &str = "verbose";
```
Comment (rivy) [inline src/uu/rm/src/rm.rs:55]: Alphabetize as you can here...

**Review by rivy — CHANGES_REQUESTED** [review]
Minor nits.

**Comment by sylvestre** [comment]
As it was only nit and the CI is green, I will land it now.

**Pull request merged**